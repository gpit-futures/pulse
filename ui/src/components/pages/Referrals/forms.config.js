export const valuesNames = {
  FROM: 'referralFrom',
  TO: 'referralTo',
  DATE: 'dateOfReferral',
  REASON: 'referralReason',
  SUMMARY: 'referralSummary',
  AUTHOR: 'author',
  DATE_CREATED: 'dateCreated',

  OUTCOME: 'referralOutcome',
  REF: 'referralRef',
  CARE_FLOW: 'referralCareFlow',
  SERVICE: 'referralServiceName',
  STATE: 'referralState',
  STATE_CODE: 'referralStateCode',
  STATE_DATE: 'referralStateDate',
  TYPE: 'referralType',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  FROM: 'Referral From',
  TO: 'Referral To',
  DATE: 'Date of Referral',
  REASON: 'Reason for Referral',
  SUMMARY: 'Clinical Summary',
  AUTHOR: 'Author',
  DATE_CREATED: 'Date',

  OUTCOME: 'Outcome of Referral',
  REF: 'Ref of Referral',
  CARE_FLOW: 'Care Flow of Referral',
  SERVICE: 'Service Name of Referral',
  STATE: 'State of Referral',
  STATE_CODE: 'State Code of Referral',
  STATE_DATE: 'State Date of Referral',
  TYPE: 'Type of Referral',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
