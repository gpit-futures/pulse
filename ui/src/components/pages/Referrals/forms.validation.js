import { valuesNames } from './forms.config';

const validateForm = (values) => {
  const errors = {};
  errors[valuesNames.FROM] = !values[valuesNames.FROM] ? 'You must enter a value.' : null;
  errors[valuesNames.TO] = !values[valuesNames.TO] ? 'You must enter a value.' : null;
  errors[valuesNames.DATE] = !values[valuesNames.DATE] ? 'You must enter a value.' : null;
  errors[valuesNames.REASON] = !values[valuesNames.REASON] ? 'You must enter a value.' : null;
  errors[valuesNames.SUMMARY] = !values[valuesNames.SUMMARY] ? 'You must enter a value.' : null;
  errors[valuesNames.AUTHOR] = !values[valuesNames.AUTHOR] ? 'You must enter a value.' : null;

  return errors
};

export { validateForm }
