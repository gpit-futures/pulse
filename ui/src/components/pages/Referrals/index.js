import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientReferralsDetailEpic } from './ducks/fetch-patient-referrals-detail.duck';
import { fetchPatientReferralsDetailEditEpic } from './ducks/fetch-patient-referrals-detail-edit.duck';
import { fetchPatientReferralsEpic } from './ducks/fetch-patient-referrals.duck';
import { fetchPatientReferralsUpdateEpic } from './ducks/fetch-patient-referrals.duck';
import { fetchPatientReferralsCreateEpic } from './ducks/fetch-patient-referrals-create.duck';

import patientsReferrals from './ducks/fetch-patient-referrals.duck';
import patientReferralsCreate from './ducks/fetch-patient-referrals-create.duck';
import referralsDetail from './ducks/fetch-patient-referrals-detail.duck';
import referralsDetailEdit from './ducks/fetch-patient-referrals-detail-edit.duck';

const epics = combineEpics(fetchPatientReferralsDetailEpic, fetchPatientReferralsDetailEditEpic, fetchPatientReferralsEpic, fetchPatientReferralsCreateEpic, fetchPatientReferralsUpdateEpic);
const Referrals = asyncComponent(() => import(/* webpackChunkName: "referrals" */ './Referrals').then(module => module.default));

const reducers = {
  patientsReferrals,
  patientReferralsCreate,
  referralsDetail,
  referralsDetailEdit,
};

const sidebarConfig = { key: 'referrals', pathToTransition: '/referrals', name: 'Referrals', isVisible: true };

const routers = [
  { key: 'referrals', component: Referrals, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.REFERRALS}` },
  { key: 'referralsCreate', component: Referrals, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.REFERRALS}/create` },
  { key: 'referralsDetail', component: Referrals, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.REFERRALS}/:sourceId` },
];

export default {
  component: Referrals,
  epics, reducers, sidebarConfig, routers,
}

