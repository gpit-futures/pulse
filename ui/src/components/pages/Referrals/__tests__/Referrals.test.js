import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import Referrals from '../Referrals';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsReferrals: {
    '9999999024': [
      {
        [valuesNames.DATE]: 1511283530634,
        [valuesNames.FROM]: 'Tony Shannon1',
        [valuesNames.TO]: 'Ripplefields Optometry service',
        [valuesNames.SOURCE]: 'ethercis',
        [valuesNames.SOURCE_ID]: '94133578-f505-4e76-b4ed-762462508801',
        dateOfReferralConvert: '21-Nov-2017',
      },
    ],
  },
};
const storeWithFormsError = mockStore(Object.assign({
  referralsDetail: {
    '9999999000': {
      [valuesNames.TYPE]: 'Referral To',
      [valuesNames.REASON]: 'Deteriorating vision',
      [valuesNames.SUMMARY]: 'Long-standing impaired visual acuity - getting much worse',
      [valuesNames.FROM]: 'Tony Shannon1',
      [valuesNames.TO]: 'Ripplefields Optometry service',
      [valuesNames.REF]: '',
      [valuesNames.OUTCOME]: '',
      [valuesNames.STATE_DATE]: 1511283530634,
      [valuesNames.STATE]: 'planned',
      [valuesNames.STATE_CODE]: 526,
      [valuesNames.CARE_FLOW]: 'Service request sent',
      [valuesNames.SERVICE]: 'Referral To',
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE]: 1511283530634,
      [valuesNames.DATE_CREATED]: 1511283530000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: '94133578-f505-4e76-b4ed-762462508801',
    },
  },
  form: {
    referralsDetailFormSelector: {
      syncErrors: {
        [valuesNames.FROM]: 'You must enter a value.',
        [valuesNames.TO]: 'You must enter a value.',
      },
    },
    referralsCreateFormSelector: {
      syncErrors: {
        [valuesNames.FROM]: 'You must enter a value.',
        [valuesNames.TO]: 'You must enter a value.',
      },
    },
  },
}, storeResource));
const storeEmpty = mockStore(Object.assign({
  referralsDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  referralsDetail: {
    '9999999000': {
      [valuesNames.TYPE]: 'Referral To',
      [valuesNames.REASON]: 'Deteriorating vision',
      [valuesNames.SUMMARY]: 'Long-standing impaired visual acuity - getting much worse',
      [valuesNames.FROM]: 'Tony Shannon1',
      [valuesNames.TO]: 'Ripplefields Optometry service',
      [valuesNames.REF]: '',
      [valuesNames.OUTCOME]: '',
      [valuesNames.STATE_DATE]: 1511283530634,
      [valuesNames.STATE]: 'planned',
      [valuesNames.STATE_CODE]: 526,
      [valuesNames.CARE_FLOW]: 'Service request sent',
      [valuesNames.SERVICE]: 'Referral To',
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE]: 1511283530634,
      [valuesNames.DATE_CREATED]: 1511283530000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: '94133578-f505-4e76-b4ed-762462508801',
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/referrals`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextCreate = generateNewContext(context, `/patients/${userId}/referrals/create`);
const contextDetail = generateNewContext(context, `/patients/${userId}/referrals/${sourceId}`);

// configuration of forms for testing methods
const formValuesEdit = {
  [valuesNames.FROM]: 'test',
  [valuesNames.TO]: 'test',
  [valuesNames.DATE]: 1511196789986,
  [valuesNames.REASON]: 'test',
  [valuesNames.SUMMARY]: 'test',
};
const formValuesCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.SOURCE_ID]: '',
  [valuesNames.FROM]: 'test',
  [valuesNames.TO]: 'test',
  [valuesNames.DATE]: '21-Dec-2017',
  [valuesNames.REASON]: 'test',
  [valuesNames.SUMMARY]: 'test',
};
const match = {
  params: {
    userId,
  },
};

describe('Component <Referrals />', () => {
  it('should renders correctly with referralPanel and testing Detail Panel', () => {
    const component = shallow(
      <Referrals
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component handleDetailReferralsClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ReferralsDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailReferralsClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: 'referralPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('ReferralsDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('referralPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('referralPanel', 'referralsDetail');
    expect(component.state().openedPanel).toEqual('referralPanel');
    expect(component.state().expandedPanel).toEqual('referralPanel');
    component.setState({ openedPanel: 'referralPanel', expandedPanel: 'referralPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'referralPanel', expandedPanel: 'all' });

    component.instance().handleExpand('referralPanel', 'referralsCreate');
    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('referralPanel', 'referralsMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('referralPanel', 'referralsMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('referralPanel', 'referralsCreate');

    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});
    component.instance().handleEdit('referralPanel');
    expect(component.state().editedPanel).toEqual({ referralPanel: true });
    component.instance().handleReferralDetailCancel('referralPanel');
    expect(component.state().editedPanel).toEqual({ referralPanel: false });
    component.instance().handleEdit('referralPanel');
    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'referralPanel');
    expect(component.state().editedPanel).toEqual({ referralPanel: false });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with referralsDetail and testing Create Panel', () => {
    const component = shallow(
      <Referrals
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ReferralsDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    // Testing component create panel methods
    component.instance().handleCreate();
    const componentStateAfterMethod = component.state();
    component.setState({ isBtnCreateVisible: false, isCreatePanelVisible: true, openedPanel: 'referralsCreate', isSecondPanel: true, isDetailPanelVisible: false, isBtnExpandVisible: false, expandedPanel: 'all', isSubmit: false, isLoading: true })
    const componentStateAfterSetState = component.state();
    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('PluginCreate')).toHaveLength(1);
    expect(component).toMatchSnapshot();
    component.instance().handleCreateCancel();
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <Referrals
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    // Testing component hideCreateForm methods
    component.instance().hideCreateForm();
    expect(component.state().isBtnCreateVisible).toEqual(true);
    expect(component.state().isCreatePanelVisible).toEqual(false);
    expect(component.state().openedPanel).toEqual('referralPanel');
    expect(component.state().isSecondPanel).toEqual(false);
    expect(component.state().expandedPanel).toEqual('all');
    expect(component.state().isBtnExpandVisible).toEqual(false);

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('dateOfReferral');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'referralFrom', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('referralFrom');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <Referrals
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextCreate);
    component.setProps({ test: 'testing create context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with forms error', () => {
    const component = shallow(
      <Referrals
        store={storeWithFormsError}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'referralPanel');
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <Referrals
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive();

    expect(component).toMatchSnapshot();
  });
});
