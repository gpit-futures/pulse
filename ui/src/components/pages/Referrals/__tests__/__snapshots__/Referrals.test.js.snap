// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Referrals /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            id="94133578-f505-4e76-b4ed-762462508801"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ReferralsDetail
          currentPanel="referralsDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "dateCreated": 1511283530000,
              "dateOfReferral": 1511283530634,
              "referralCareFlow": "Service request sent",
              "referralFrom": "Tony Shannon1",
              "referralOutcome": "",
              "referralReason": "Deteriorating vision",
              "referralRef": "",
              "referralServiceName": "Referral To",
              "referralState": "planned",
              "referralStateCode": 526,
              "referralStateDate": 1511283530634,
              "referralSummary": "Long-standing impaired visual acuity - getting much worse",
              "referralTo": "Ripplefields Optometry service",
              "referralType": "Referral To",
              "source": "ethercis",
              "sourceId": "94133578-f505-4e76-b4ed-762462508801",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="referralsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="referralsCreate"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with forms error 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with referralPanel and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            id="94133578-f505-4e76-b4ed-762462508801"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ReferralsDetail
          currentPanel="referralsDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "dateCreated": 1511283530000,
              "dateOfReferral": 1511283530634,
              "referralCareFlow": "Service request sent",
              "referralFrom": "Tony Shannon1",
              "referralOutcome": "",
              "referralReason": "Deteriorating vision",
              "referralRef": "",
              "referralServiceName": "Referral To",
              "referralState": "planned",
              "referralStateCode": 526,
              "referralStateDate": 1511283530634,
              "referralSummary": "Long-standing impaired visual acuity - getting much worse",
              "referralTo": "Ripplefields Optometry service",
              "referralType": "Referral To",
              "source": "ethercis",
              "sourceId": "94133578-f505-4e76-b4ed-762462508801",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="referralsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="referralPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with referralPanel and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ReferralsDetail
          currentPanel="referralsDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "dateCreated": 1511283530000,
              "dateOfReferral": 1511283530634,
              "referralCareFlow": "Service request sent",
              "referralFrom": "Tony Shannon1",
              "referralOutcome": "",
              "referralReason": "Deteriorating vision",
              "referralRef": "",
              "referralServiceName": "Referral To",
              "referralState": "planned",
              "referralStateCode": 526,
              "referralStateDate": 1511283530634,
              "referralSummary": "Long-standing impaired visual acuity - getting much worse",
              "referralTo": "Ripplefields Optometry service",
              "referralType": "Referral To",
              "source": "ethercis",
              "sourceId": "94133578-f505-4e76-b4ed-762462508801",
            }
          }
          editedPanel={Object {}}
          expandedPanel="referralPanel"
          isSubmit={false}
          name="referralsDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="referralPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with referralPanel and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with referralsDetail and testing Create Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="referralsCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="referralsCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="referralsCreate"
          title="Create Referral"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Referrals /> should renders correctly with referralsDetail and testing Create Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="referralsMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="referralsMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Referrals"
          />
          <PluginMainPanel
            columnNameSortBy="dateOfReferral"
            emptyDataMessage="No referrals"
            headers={
              Array [
                Object {
                  "key": "dateOfReferral",
                  "title": "Date of Referral",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "referralFrom",
                  "title": "Referral From",
                  "width": "28%",
                },
                Object {
                  "key": "referralTo",
                  "title": "Referral To",
                  "width": "28%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "SourceID",
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            table="referrals"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="referralsCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="referralsCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          openedPanel="referralsCreate"
          title="Create Referral"
        />
      </Col>
    </Row>
  </div>
</section>
`;
